packages/flutter/lib/src/material/slider.dart

```diff
@@ -1950,37 +1974,83 @@ class _SliderDefaultsM2 extends SliderThemeData {
   Color? get disabledThumbColor => Color.alphaBlend(_colors.onSurface.withOpacity(.38), _colors.surface);
 
   @override
-  Color? get overlayColor => _colors.primary.withOpacity(0.12);
+  Color? get overlayColor {
+    if (_sliderTheme.thumbShape is BarSliderThumbShape) {
+      return Colors.transparent;
+    }
+    return _colors.primary.withOpacity(0.12);
+  }
 
   @override
   TextStyle? get valueIndicatorTextStyle => Theme.of(context).textTheme.bodyLarge!.copyWith(
-    color: _colors.onPrimary,
+    color: _sliderTheme.valueIndicatorShape is RoundedRectSliderValueIndicatorShape
+      ? _colors.onInverseSurface
+      : _colors.onPrimary,
   );
 
   @override
   SliderComponentShape? get valueIndicatorShape => const RectangularSliderValueIndicatorShape();
-}
 
-// TODO(quncheng): Update M3 defaults to match the latest specs.
-// BEGIN GENERATED TOKEN PROPERTIES - Slider
+  @override
+  SliderComponentShape? get thumbShape => const RoundSliderThumbShape();
 
-// Do not edit by hand. The code between the "BEGIN GENERATED" and
-// "END GENERATED" comments are generated from data in the Material
-// Design token database by the script:
-//   dev/tools/gen_defaults/bin/gen_defaults.dart.
+  @override
+  SliderTrackShape? get trackShape => const RoundedRectSliderTrackShape();
 
-class _SliderDefaultsM3 extends SliderThemeData {
-  _SliderDefaultsM3(this.context)
-    : super(trackHeight: 4.0);
+  @override
+  SliderTickMarkShape? get tickMarkShape => const RoundSliderTickMarkShape();
+
+  @override
+  SliderComponentShape? get overlayShape => const RoundSliderOverlayShape();
+
+  @override
+  Color? get valueIndicatorColor {
+    if (_sliderTheme.valueIndicatorShape is RoundedRectSliderValueIndicatorShape) {
+      return _colors.inverseSurface;
+    }
+    return _colors.primary;
+  }
+
+  @override
+  double? get trackHeight => _sliderTheme.trackShape is GappedSliderTrackShape
+    ? 16.0
+    : 4.0;
+
+  @override
+  MaterialStateProperty<Size?>? get barThumbSize =>
+    MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.pressed)) {
+        return  const Size(2.0, 44.0);
+      }
+      return  const Size(4.0, 44.0);
+    });
+
+  @override
+  double? get trackGapSize => 6.0;
+}
+
+/// The default [SliderThemeData] for the legacy Material 3 slider.
```

Comment:
nit: please clarify that legacy = first version of the M3 slider and that the look has been updated this year

Reply:
Updated